File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        unordered_map<int,int>mp; 
        for(int i=0;i<nums.size();i++)
        {
            mp[nums[i]]++;
        } int k=1;
        for(int i=0;i<nums.size();i++)
        {
            if(mp.find(k)==mp.end()) return k; 
            k++;
        }
        return nums.size()+1;
    }
};

File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        HashSet<Integer> st=new HashSet<>(); 
        for(int i:nums)
        {
            st.add(i); 
        }
        int n=1;
        while(true)
        {
            if(!st.contains(n))
            {
                return n; 
            }
            n++;
        }
    }
}